pkg/controllers/jobset_controller.go

```diff
@@ -913,3 +936,88 @@ func findReplicatedStatus(replicatedJobStatus []jobset.ReplicatedJobStatus, repl
 	}
 	return jobset.ReplicatedJobStatus{}
 }
+
+func (r *JobSetReconciler) deleteJobSet(ctx context.Context, js *jobset.JobSet) error {
+	log := ctrl.LoggerFrom(ctx)
+
+	policy := metav1.DeletePropagationForeground
+	options := []client.DeleteOption{client.PropagationPolicy(policy), client.Preconditions{UID: &js.UID}}
```

Comment:
Do we need the jobset UID as a precondition here, since the UID is already part of the jobset object being passed to `Delete()`?

Please correct me if I'm wrong but I thought we would only need to pass the UID as a precondition if we were deleting by name or by namespaced-name or something that may not be unique over time (e.g., same JobSet rapidly created/deleted could lead to race condition).

Reply:
I guess not, I will remove it